esmvalcore/preprocessor/_area.py

```diff
@@ -331,14 +331,23 @@ def _crop_cube(cube, start_longitude, start_latitude, end_longitude,
         lon_bound = lon_coord.core_bounds()[0]
         lon_step = lon_bound[1] - lon_bound[0]
         start_longitude -= lon_step
+        if start_longitude < -180:
+            start_longitude = -180.
         end_longitude += lon_step
+        if end_longitude > 360:
+            end_longitude = 360.
+        if end_longitude > 180 and start_longitude == -180.:
+            end_longitude = 180.
```

Comment:
This looks a bit worrying, is this needed?

Reply:
yes, since the padding for files that are full maps goes over 180, specifically needed for this case

Reply:
Maybe you could take this out of this pull request and add it in the other pull request where you add support for shapefiles with longitude coordinates between -180 and 180? At the moment these aren't supported anyway.

Reply:
On line 334 and 335 there is also code related to supporting those files, maybe it would be best to also move that to the other pull request. I think it does not work in all cases when you make it like this, so it would be better to just get this pull request merged with things we know do work and try to add a new feature in the next pull request.

Reply:
> Maybe you could take this out of this pull request and add it in the other pull request where you add support for shapefiles with longitude coordinates between -180 and 180? At the moment these aren't supported anyway.

where's the restriction clause? I can't see it and the thing fails if any of these files is used - the user doesn't know about this restriction and will attempt to use whatevs shapefile that may get into this trouble

Reply:
it's perfectly fine to have this check here since it doesn't affect any other case imho

Reply:
The problem is that it doesn't work if you do it like this. To make it work, you first need to distinguish between the two cases 1) longitude in domain 0 to 360 or 2) longitude in domain -180 to 180 and then limit the `start_longitude` and `end_longitude` arguments you provide to `extract_shape` to the chosen domain.

Reply:
`start_longitude == -180.` implies we are on a [-180, 180] grid, mate - we can not be on a [0, 360] and if lons go from [-180, 360] then it means we're wrapping coords like Xmas presents and I am a fish :fish: - both unlikely :grin:

Reply:
Yes, but a start longitude of say -170 would probably also mean a  [-180, 180] grid, but your code does not cover that case.

Reply:
ok @bouweandela you win :checkered_flag: - I have removed that special case and put it in https://github.com/ESMValGroup/ESMValCore/pull/319 where there is an actual checking mechanism for these sort of files :beer:

Reply:
:beers: